src/compiler/checker.ts

```diff
@@ -47488,6 +47488,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function checkSourceElementWorker(node: Node): void {

```

Comment:
I think this function (and checkSourceElement) should probably only accept SourceElement types rather than Node.

Reply:
I just tried that and it causes a million errors, some because if you write `SourceElement` as a union of specific node types, then `TypeNode` is not assignable to `SourceElement`, and some others because `forEachChild`'s type doesn't know the children of, say, `ClassDeclaration` are all `SourceElement`. So I'm not doing that here.